bot/exts/moderation/clean.py

```diff
@@ -464,13 +464,13 @@ async def _clean_messages(
     async def clean_group(
         self,
         ctx: Context,
-        users: Greedy[User] = None,
+        users: Greedy[User] = None,  # noqa: RUF013
         first_limit: CleanLimit | None = None,
         second_limit: CleanLimit | None = None,
         regex: Regex | None = None,
         bots_only: bool | None = False,
         *,
-        channels: CleanChannels = None  # "Optional" with discord.py silently ignores incorrect input.
+        channels: CleanChannels | None = None  # "Optional" with discord.py silently ignores incorrect input.
```

Comment:
Comment here suggests this may not be correct, and probably the same for some of the other changes (I assume discord.py treats `| None` the same way as `Optional`). Not sure exactly what the difference in behaviour would be though.

Reply:
I tested with a few commands, and it seemed to have the same behaviour. haven't tested all of the commands though, and I'm not aware of edge cases that might need changing. Can look further into this another day unless someone gets there first.

Reply:
I think it makes a difference with keyword only arguments: Without `| None` you will get an error for anything that doesn't match the converter, e.g. `!clean bots 1121140691930726520 a b c` says `Channel "a" not found.`. With `| None` it is just ignored so you don't get the error.

Reply:
Ah, some more testing, it matters for any args, not just keyword only. If there's a convertor error it just gets ignored.

Updated in https://github.com/python-discord/bot/compare/af28b6d522d6a38d35dda460076539a77b5d967f..a3b688bdcff0338d87e9c8e9255f7944f869d37b

Reply:
> Updated in https://github.com/python-discord/bot/compare/af28b6d522d6a38d35dda460076539a77b5d967f..a3b688bdcff0338d87e9c8e9255f7944f869d37b

It looks like there are still some in `clean.py`, `management.py` and `silence.py` that need fixing.

Reply:
Ah yea, those should be fixed now.